src/features/click-to-load.js

```diff
@@ -110,7 +110,30 @@ class DuckWidget {
                     // with a light version, replace with full version.
                     this.clickAction.type = 'allowFull'
                 }
-                value = attrSettings.default
+                // Our default value for width is often greater than the window size of smaller
+                // screens (ie mobile). When it is, use either the parent width (if reliable)
+                // or the window width instead.
+                if (attrName === 'data-width' && attrSettings.default > window.innerWidth) {
+                    const windowWidth = window.innerWidth
+                    const parentElement = this.originalElement.parentElement
+                    const parentStyles = parentElement
+                        ? window.getComputedStyle(parentElement)
+                        : null
+                    // We want to calculate the inner width of the parent element as the iframe, when added back,
+                    // should not be bigger than the space available in the parent element. There is no straightforward way of
+                    // doing this. We need to get the parent's .clientWidth and remove the paddings size from it.
+                    const parentInnerWidth = parentElement && parentStyles && parentStyles.display !== 'inline'
```

Comment:
I recommend against mixing ternary operator with logical and/or. Sometimes the results can be surprising. If you want to stick with this, I recommend at least adding parenthesis around the condition.
```suggestion
                    const parentInnerWidth = (parentElement && parentStyles && parentStyles.display !== 'inline')
```